src/libraries/System.Composition.Runtime/src/System/Composition/Hosting/Core/CompositionContract.cs

```diff
@@ -45,7 +45,7 @@ public CompositionContract(Type contractType, string contractName)
         public CompositionContract(Type contractType, string contractName, IDictionary<string, object> metadataConstraints)
         {
             if (contractType == null) throw new ArgumentNullException(nameof(contractType));
-            if (metadataConstraints?.Count == 0) throw new ArgumentOutOfRangeException(nameof(metadataConstraints));
+            if (metadataConstraints == null || metadataConstraints.Count == 0) throw new ArgumentOutOfRangeException(nameof(metadataConstraints));
```

Comment:
This case is incorrect. Nulls are explicitly accepted from other constructors. Note that nullable annotations are NOT enabled for this project.